pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -1196,3 +1197,15 @@ func KubernetesDashboardEnabled(addons *gardencorev1beta1.Addons) bool {
 func NginxIngressEnabled(addons *gardencorev1beta1.Addons) bool {
 	return addons != nil && addons.NginxIngress != nil && addons.NginxIngress.Enabled
 }
+
+// ConstantConditionPatch creates a strategic merge patch with the the given conditions.
+func ConstantConditionPatch(conditions ...gardencorev1beta1.Condition) (client.Patch, error) {
```

Comment:
Does it make sense to have the patch type as a parameter instead of hard-coding it to strategic-merge-patch?

```suggestion
func ConstantConditionPatch(patchType types.PatchType, conditions ...gardencorev1beta1.Condition) (client.Patch, error) {
```